broker-daemon/models/block-order.js

```diff
@@ -160,7 +160,24 @@ class BlockOrder {
    */
   fail () {
     this.status = BlockOrder.STATUSES.FAILED
+    return this
+  }
+
+  /**
+   * Move the block order to a completed status
+   * @return {BlockOrder} Modified block order instance
+   */
+  complete () {
+    const activeOrders = this.openOrders.filter(o => o.status !== BlockOrder.STATUSES.COMPLETED)
+    const activeFills = this.fills.filter(f => f.status !== BlockOrder.STATUSES.COMPLETED)
+
+    // If we still have active orders or fills, then the block order is not completed
+    // so we can simply return the object
+    if (activeOrders || activeFills) {
+      return this
```

Comment:
It seems odd to not error out if we aren't updating the status. Why do that?

Reply:
This is a future change. If the block order still has active orders/fills we dont want to update its status to completed